compute/cker/src/train/MaxPool.test.cc

```diff
@@ -0,0 +1,279 @@
+    input_deriv_mat << /* depth0 */ 0, 0, 0, 0, 0, 0, 0.5, 0, 0,
+      /* depth1 */ 0, 0, 0, 0, 0.9, 0, 0, 0, 0;
+    verifier.verifyBackward(output_deriv, expected_input_deriv);
+  }
+
+  // with padding case
+  {
+    nnfw::cker::PoolParams op_param;
+    {
+      op_param.stride_height = 2;
+      op_param.stride_width = 2;
+      op_param.filter_height = 2;
+      op_param.filter_width = 2;
+      op_param.padding_values.height = 2;
+      op_param.padding_values.width = 2;
+    }
+    nnfw::cker::Shape in = {1, 2, 2, 1};
+    nnfw::cker::Shape out = {1, 3, 3, 1};
+
+    MaxPoolOpVerifier<float> verifier(op_param, in, out);
+
+    /**
+     * input_with_padding:             expected_output:
+     *
+     *    4   8                              0  0  0
+     *    9   2            -(forward)->      0  9  0
+     *                                       0  0  0
+     */
+
+    std::vector<float> input = {4, 8, 9, 2};
+    std::vector<float> expected_output = {0, 0, 0, 0, 9, 0, 0, 0, 0};
+    verifier.verifyForward(input, expected_output);
+
+    /**
+     * output_deriv:                    input_deriv:
+     *
+     *  0.1   0.1   0.1                     0     0
+     *  0.1   0.2   0.3   -(backward)->     0.2   0
+     *  0.5   0.1   0.1
+     */
+    std::vector<float> output_deriv = {0.1, 0.1, 0.1, 0.1, 0.2, 0.3, 0.5, 0.1, 0.1};
+    std::vector<float> expected_input_deriv = {0, 0, 0.2, 0};
+    verifier.verifyBackward(output_deriv, expected_input_deriv);
+  }
+}
+
+TEST(CKer_Operation, neg_MaxPool)
+{
+  // Invalid expected value
+  {
+    nnfw::cker::PoolParams op_param;
+    {
+      op_param.stride_height = 1;
+      op_param.stride_width = 1;
+      op_param.filter_height = 2;
+      op_param.filter_width = 2;
+      op_param.padding_values.height = 0;
+      op_param.padding_values.width = 0;
+    }
+    nnfw::cker::Shape in = {1, 2, 2, 1};
+    nnfw::cker::Shape out = {1, 1, 1, 1};
+
+    MaxPoolOpVerifier<float> verifier(op_param, in, out);
+
+    std::vector<float> input = {0, 0, 0, 0};
+    std::vector<float> expected_output = {-1};
+
+    verifier.verifyForward(input, expected_output, false);
+  }
+
+  // Invalid expected value
+  {
+    nnfw::cker::PoolParams op_param;
+    {
+      op_param.stride_height = 2;
+      op_param.stride_width = 2;
+      op_param.filter_height = 2;
+      op_param.filter_width = 2;
+      op_param.padding_values.height = 1;
+      op_param.padding_values.width = 1;
+    }
+
+    nnfw::cker::Shape in = {1, 2, 2, 1};
+    nnfw::cker::Shape out = {1, 2, 2, 1};
+
+    MaxPoolOpVerifier<float> verifier(op_param, in, out);
+
+    std::vector<float> input = {0, 0, 0, 0};
+    std::vector<float> expected_output = {0, 0, 0, 0};
+    verifier.verifyForward(input, expected_output);
+
+    std::vector<float> output_deriv = {0.1, 0.1, 0.1, 0.2};
+    std::vector<float> expected_input_deriv = {0.1, 0.1, 0.1, 0.1};
+    verifier.verifyBackward(output_deriv, expected_input_deriv, false);
+  }
```

Comment:
(lately updated) Here, I updated tc with padding and negative one. (https://github.com/Samsung/ONE/pull/11686#issuecomment-1757001754)